misc/userscripts/password_fill

```diff
@@ -111,17 +111,18 @@ simplify_url() {
 # Another behavior is to drop another level of subdomains until search hits
 # are found:
 no_entries_found() {
-    while [ 0 -eq "${#files[@]}" ] && [ -n "$simple_url" ]; do
+    temp_simple_url=$simple_url
+    while [ 0 -eq "${#files[@]}" ] && [ -n "$temp_simple_url" ]; do
         # shellcheck disable=SC2001
-        shorter_simple_url=$(sed 's,^[^.]*\.,,' <<< "$simple_url")
-        if [ "$shorter_simple_url" = "$simple_url" ] ; then
+        shorter_simple_url=$(sed 's,^[^.]*\.,,' <<< "$temp_simple_url")
+        if [ "$shorter_simple_url" = "$temp_simple_url" ] ; then
             # if no dot, then even remove the top level domain
-            simple_url=""
-            query_entries "$simple_url"
+            temp_simple_url=""
+            query_entries "$temp_simple_url"
             break
         fi
-        simple_url="$shorter_simple_url"
-        query_entries "$simple_url"
+        temp_simple_url="$shorter_simple_url"
+        query_entries "$temp_simple_url"
         #die "No entry found for »$simple_url«"
         # enforce menu if we do "fuzzy" matching
         menu_if_one_entry=1
```

Comment:
It's a good thing to use another variable name here. One can even write:
```
   local temp_simple_url="$1"
```
I'm not sure whether it was a mistake of mine that `$simple_url` is used instead of `$1`; maybe I intended to allow `no_entries_found` to simplify the url further, but I can't remember at the moment.